vip-jetpack.php

```diff
@@ -28,5 +28,12 @@
  * On VIP Go, we always want to use the Go Photon service, instead of WordPress.com's
  */
 add_filter( 'jetpack_photon_domain', function( $domain, $image_url ) {
-	return home_url();
+	// only apply to photon images
+	if ( strpos( $domain, 'wp.com' ) !== false) {
```

Comment:
It might be safer to match with regex. Jetpack says:

> http://i{$subdomain}.wp.com Domain used by Photon. $subdomain is a random number between 0 and 2.

Possible regex:

```php
if ( preg_match( '|^https?://i[0-9]+.wp.com$|', $domain ) ) {
```

Matches:

* Domain needs to start with `http://i` or `https://i`
* The domain needs to end with `.wp.com`
* There needs to be a series of one or more numbers in the middle

What do you reckon?

From my testing:

```
 Marker - 25 Jan 2016, 16:02:08
[25-Jan-2016 16:02:11 UTC] PASSED: http://i1.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: http://i5.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: http://i3.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: http://i999.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: http://i0.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: https://i1.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: https://i5.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: https://i3.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: https://i999.wp.com
[25-Jan-2016 16:02:11 UTC] PASSED: https://i0.wp.com
[25-Jan-2016 16:02:11 UTC] FAILED: whatever-http://i3.wp.com
[25-Jan-2016 16:02:11 UTC] FAILED: http://i3.wp.com-whatever
[25-Jan-2016 16:02:11 UTC] FAILED: http://isethwp.com
[25-Jan-2016 16:02:11 UTC] FAILED: http://example.com
[25-Jan-2016 16:02:11 UTC] FAILED: https://isethwp.com
[25-Jan-2016 16:02:11 UTC] FAILED: https://example.com
[25-Jan-2016 16:02:11 UTC] FAILED: s
 ```